tests/components/modbus/test_sensor.py

```diff
@@ -555,6 +556,35 @@ async def test_all_sensor(hass, cfg, regs, expected):
             [0x0101],
             "257",
         ),
+        (
+            {
+                CONF_COUNT: 1,
+                CONF_REGISTER_SIZE: 4,
+                CONF_DATA_TYPE: DATA_TYPE_CUSTOM,
```

Comment:
It also needs to be tested with DATA_TYPE_INT etc. since the validator allows this.

Reply:
This is not resolved try e.g. CONF_STRUCTURE: “>i”.

Reply:
How do you solve when CONF_STRUCTURE request 2 bytes, but register_size is 4 ? According to the validator that is a legal configuration.